mne/preprocessing/tests/test_maxwell.py

```diff
@@ -1118,5 +1137,34 @@ def test_find_bad_channels_maxwell(fname, bads, annot, add_ch, ignore_ref,
     assert 'Interval   1:    0.00' in log
     assert 'Interval   2:    5.00' in log
 
+    if return_scores:
+        meg_chs = raw.copy().pick_types(meg=True, exclude=[]).ch_names
+        ch_types = raw.get_channel_types(meg_chs)
+
+        assert list(got_scores['ch_names']) == meg_chs
+        assert list(got_scores['ch_types']) == ch_types
+
+        assert (got_scores['scores_flat'].shape ==
+                got_scores['limits_flat'].shape ==
+                got_scores['scores_noisy'].shape ==
+                got_scores['limits_noisy'].shape ==
+                (len(meg_chs), len(got_scores['bin_edges']) - 1))
+
+        for want_bad in want_bads:
+            ch_idx = list(got_scores['ch_names']).index(want_bad)
+            scores = got_scores['scores_noisy'][ch_idx]
+            limits = got_scores['limits_noisy'][ch_idx]
+            n_segments_exceed_limit = (scores > limits).sum()
+            assert (n_segments_exceed_limit >=
+                    min(min_count, len(got_scores['bin_edges']) - 1))
```

Comment:
Rather than iterate over the ones it should find, why not just redo the computation with something like:
```
n_segments_exceed_limit = (got_scores['scores_noisy'] >= got_scores['limits_noisy']).sum(-1)
new_bad = set(got_scores['ch_names'][idx] for idx in np.where(n_segments_exceed_limit >= min(...))
assert new_bad == set(want_bads)
```
It checks both hits and correct rejections rather than just hits.

Reply:
I have now implemented your suggestion in 694a8b2